include/pybind11/async.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include <memory>
+#include <future>
+#include <map>
+#include <algorithm>
+#include <sstream>
+
+#include <chrono>
+#include <iostream>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+#include "pybind11/embed.h"
+#include "pybind11/operators.h"
+
+
+namespace pybind11 {
+
+template<typename ResultType>
+class Awaitable : public std::enable_shared_from_this<Awaitable<ResultType>>{
```

Comment:
How does pybind11 know how to map this to Python?

Reply:
But OK, yes. Maybe this could become some kind of custom caster or convenience macro or so.

Reply:
thx @YannickJadoul !

so basically you are suggesting to have some sort of a .def(_, _, py::async()) argument?
and some sort of py::enable_async() macro for the module definition?